pages/map/index.tsx

```diff
@@ -1,6 +1,148 @@
+import Head from 'next/head';
+import { useState } from 'react';
+import {Container, Row, Col, Form, Button} from 'react-bootstrap';
+import {BiCurrentLocation, BiSearchAlt, BiStreetView} from 'react-icons/bi';
+import GoogleMapReact from 'google-map-react';
+
+import Footer from '../../components/footer';
+import Navigation from '../../components/navbar';
+
+const mapsKey: any = process.env.NEXT_PUBLIC_GOOGLEMAPS_API_KEY;
+
+interface Coords{
+    lat: number,
+    lng: number
+}
+
+interface GMap{
+    center: Coords
+    children?: React.ReactNode
+}   
+
+interface MapPageContent{
+    children: React.ReactNode
+}
+
+const GMap = ({center, children}: GMap) => {
+    //Creates Google Map
+    const defaultCoords: Coords = {lat: 40.7128, lng: -74.0060}; //Defaults to NYC coords
+    const mapOptions = {
+        mapTypeControl: true,
+        streetViewControl: true
+    }
+
+    return(
+        <div className='shadow' style={{width: "100%", height: "80vh"}}>
+            {center.lat && center.lng ? 
+                <GoogleMapReact
+                    yesIWantToUseGoogleMapApiInternals
+                    bootstrapURLKeys={{key: mapsKey, libraries:[]}}
+                    defaultCenter={defaultCoords}
+                    center={center}
+                    defaultZoom={11}
+                    options={mapOptions}
+                >
+                    <BiStreetView
+                        //@ts-ignore
```

Comment:
curious, why does this need `@ts-ignore` comment? what's the error that shows up? is it because of a type mismatch?

Reply:
`Type '{ lat: number; lng: number; text: string; size: number; style: { position: "absolute"; transform: "translate(-50%, -50%)"; }; }' is not assignable to type 'IntrinsicAttributes & IconBaseProps'.
  Property 'lat' does not exist on type 'IntrinsicAttributes & IconBaseProps'.`

Most likely due to google-map-react. In order to display something on the map, in this case, this icon, representing a user's location, google-map-react requires lat and lng props to be passed to components to be rendered on the map. Since these props don't exist on icons, TS raises the above error.